src/serac/physics/integrators/displacement_hyperelastic_integrator.hpp

```diff
@@ -65,24 +67,97 @@ class IncrementalHyperelasticIntegrator : public mfem::NonlinearFormIntegrator {
                                    const mfem::Vector& elfun, mfem::DenseMatrix& elmat);
 
 private:
+  /**
+   * @brief Calculate the deformation gradient and right Cauchy-Green deformation tensor at a quadrature point
+   *
+   * @param[in] el The finite element
+   * @param[in] ip The integration point
+   * @param[in] Ttr The reference-to-target (stress-free) transformation
+   */
+  void CalcDeformationGradient(const mfem::FiniteElement& el, const mfem::IntegrationPoint& ip,
+                               mfem::ElementTransformation& Ttr);
+
   /**
    * @brief The associated hyperelastic model
    */
-  mfem::HyperelasticModel* model_;
+  serac::HyperelasticMaterial& material_;
+
+  /**
+   * @brief gradients of reference shape functions (dof x dim).
+   *
+   */
+  mfem::DenseMatrix DSh_;
+
+  /**
+   * @brief gradients of the shape functions in the target configuration (DN_i, DX_j)
+   *
+   */
+  mfem::DenseMatrix DS_;
+
+  /**
+   * @brief gradients of the shape functions in the current configuration (DN_i, Dx_j)
+   *
+   */
+  mfem::DenseMatrix B_;
+
+  /**
+   * @brief the Jacobian of the target-to-reference-element transformation.
+   *
+   */
+  mfem::DenseMatrix Jrt_;
+
+  /**
+   * @brief the deformation gradient
+   *
+   */
+  mfem::DenseMatrix F_;
+
+  /**
+   * @brief the inverse of the deformation gradient
+   *
+   */
+  mfem::DenseMatrix Finv_;
+
+  /**
+   * @brief the spatial tangent moduli
+   *
+   */
+  mfem_ext::Array4D<double> C_;
+
+  /**
+   * @brief the Cauchy stress
+   *
+   */
+  mfem::DenseMatrix sigma_;
+
+  /**
+   * @brief Current input state dofs (dof x dim)
+   *
+   */
+  mfem::DenseMatrix PMatI_;
```

Comment:
these are nodal displacements? what does the name `PMatI_` mean?

Reply:
I'm assuming `dofs` is the number of dofs per element, so it's an element-local thing

Reply:
Yes - this old form of the nonlinear form integrator is called on an element-by-element basis.